test/integration/pod_node_constraints_test.go

```diff
@@ -0,0 +1,115 @@
+	pluginapi "github.com/openshift/origin/pkg/scheduler/admission/podnodeconstraints/api"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+)
+
+func testPodNodeConstraintsPod(nodeName string, nodeSelector *map[string]string) *kapi.Pod {
+	pod := &kapi.Pod{}
+	pod.Name = "testpod"
+	pod.Spec.RestartPolicy = kapi.RestartPolicyNever
+	if len(nodeName) > 0 {
+		pod.Spec.NodeName = nodeName
+	}
+	if len(*nodeSelector) > 0 {
+		pod.Spec.NodeSelector = *nodeSelector
+	}
+	pod.Spec.Containers = []kapi.Container{
+		{
+			Name:  "container",
+			Image: "test/image",
+		},
+	}
+	return pod
+}
+
+func testPodNodeConstraintsExpectedError(errorMsg string) error {
+	attrs := admission.NewAttributesRecord(testPodNodeConstraintsPod("", &map[string]string{}), kapi.Kind("Pod"), testutil.Namespace(), "test", kapi.Resource("pods"), "", admission.Create, nil)
+	return admission.NewForbidden(attrs, fmt.Errorf(errorMsg))
+}
+
+func TestPodNodeConstraintsAdmissionPluginDefaults(t *testing.T) {
+	config := &pluginapi.PodNodeConstraintsConfig{}
+	kclient := setupPodNodeConstraintsTest(t, config)
+	nodeSelector := &map[string]string{"bogus": "frank"}
+	_, err := kclient.Pods(testutil.Namespace()).Create(testPodNodeConstraintsPod("nodename.example.com", nodeSelector))
+	if err != nil {
+		t.Fatalf("Unexpected: %v", err)
+	}
+}
+
+func TestPodNodeConstraintsAdmissionPluginProhibitNodeName(t *testing.T) {
+	config := &pluginapi.PodNodeConstraintsConfig{
+		ProhibitNodeName: true,
+	}
+	kclient := setupPodNodeConstraintsTest(t, config)
+	expectedError := testPodNodeConstraintsExpectedError("use of NodeName field in pod spec prohibited by policy")
+	nodeSelector := &map[string]string{"bogus": "frank"}
+	pod, err := kclient.Pods(testutil.Namespace()).Create(testPodNodeConstraintsPod("nodename.example.com", nodeSelector))
+	if err == nil {
+		t.Fatalf("Expected error %q, no error received; pod: %#v", expectedError.Error(), pod)
+	}
+	if err.Error() != expectedError.Error() {
+		t.Errorf("expected error %q, got: %q; pod: %#v", expectedError.Error(), err.Error(), pod)
+	}
+}
+
+func TestPodNodeConstraintsAdmissionPluginProhibitNodeSelector(t *testing.T) {
+	config := &pluginapi.PodNodeConstraintsConfig{
+		ProhibitNodeSelector: true,
+	}
+	kclient := setupPodNodeConstraintsTest(t, config)
+	expectedError := testPodNodeConstraintsExpectedError("use of NodeSelector field in pod spec prohibited by policy")
+	nodeSelector := &map[string]string{"bogus": "frank"}
+	_, err := kclient.Pods(testutil.Namespace()).Create(testPodNodeConstraintsPod("nodename.example.com", nodeSelector))
+	if err == nil {
+		t.Fatalf("Expected error %q, no error received", expectedError.Error())
+	}
+	if err.Error() != expectedError.Error() {
+		t.Errorf("expected error %q, got: %q", expectedError.Error(), err.Error())
+	}
+}
+
+func setupPodNodeConstraintsTest(t *testing.T, pluginConfig *pluginapi.PodNodeConstraintsConfig) kclient.Interface {
```

Comment:
This is getting stubbed pretty often for these admission plugins.  Open an issue to refactor into a common helper with a better interface.  I could see an argument for splitting our integration tests too, but that'll be for another time.